### PR TITLE
companion for #12663 jsonrpsee v0.16

### PR DESCRIPTION
**Pull request by niklasad1**
Companion for https://github.com/paritytech/substrate/pull/12663

cumulus companion: https://github.com/paritytech/cumulus/pull/1905

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2079305

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2079525

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2100468

**Comment by lexnv** [comment]
bot rebase

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2112302

File: rpc/Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ authors.workspace = true
 edition.workspace = true
 
 [dependencies]
-jsonrpsee = { version = "0.15.1", features = ["server"] }
+jsonrpsee = { version = "0.16.1", features = ["server"] }
```
Comment (lexnv) [inline rpc/Cargo.toml:8]:
```suggestion
jsonrpsee = { version = "0.16.2", features = ["server"] }
```

File: utils/staking-miner/Cargo.toml [hunk 2]
```diff
@@ -8,7 +8,7 @@ edition.workspace = true
 codec = { package = "parity-scale-codec", version = "3.0.0" }
 clap = { version = "4.0.9", features = ["derive", "env"] }
 tracing-subscriber = { version = "0.3.11", features = ["env-filter"] }
-jsonrpsee = { version = "0.15.1", features = ["ws-client", "macros"] }
+jsonrpsee = { version = "0.16.1", features = ["ws-client", "macros"] }
```
Comment (lexnv) [inline utils/staking-miner/Cargo.toml:11]:
```suggestion
jsonrpsee = { version = "0.16.2", features = ["ws-client", "macros"] }
```

**Review by lexnv — APPROVED** [review]
LGTM! Small changes of version to `0.16.2` to keep consistency with substrate, looks good! 👍

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2140110

**Review by ordian — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-36/1529/1

**Pull request merged**